framework/util/argument_parser.cpp

```diff
@@ -245,7 +246,8 @@ void ArgumentParser::Init(std::vector<std::string> command_line_args,
                             }
                             argument_values_[cur_argument.second] = argument_value;
                         }
-                        is_argument = true;
+                        arguments_present_[cur_argument.second] = true;
```

Comment:
Should this be "false" for when the argument is invalid, like checked above on line 266?

Reply:
I think this might need to be bumped up one line so it's inside the `else`

Reply:
It seems to me that since we can check for validity with `IsInvalid()`, `IsArgumentSet()` should indicate whether or not the user passed the given argument regardless of validity.

Reply:
I think I'd argue the opposite.  Was `ArgumentSet` (i.e. is `_arguments_present_`) if the option occurred at the end of the command line and there was no filename for `--output`?

In any case, It doesn't have to be in `arguments_present_` if `is_invalid_` since all the tools exit if `IsInvalid()` before checking the arguments and `GetInvalidArgumentOrOptions()` would return the invalid arguments.  (No tool calls `GetInvalidArgumentOrOptions `currently, they just print Usage and exit, but I think let's leave that for another PR!)

Reply:
It doesn't look like you made the change I requested?

Reply:
My bad...it's pushed now